udata/models/extras_fields.py

```diff
@@ -60,3 +60,27 @@ def to_python(self, value):
         if isinstance(value, EmbeddedDocument):
             return value
         return super(ExtrasField, self).to_python(value)
+
+
+class OrganizationExtrasField(ExtrasField):
+    def __init__(self, **kwargs):
+        super(OrganizationExtrasField, self).__init__()
+
+    def validate(self, values):
+        super(ExtrasField, self).validate(values)
+
+        errors = {}
+
+        expected_keys = ["title", "description", "type", "choices"]
+        valid_types = ["str", "int", "float", "bool", "datetime", "date", "choice"]
+
+        for elem in values.get('custom', []):
+            # Check if the dictionary contains the expected keys and only them
+            if all(key in expected_keys for key in elem.keys()):
+                if elem.get("type") not in valid_types:
+                    errors['type'] = 'Value should be one of: {types}'.format(types=valid_types)
```

Comment:
Maybe add more context to the error.
```suggestion
                    errors['type'] = 'Type \'{type}\' of \'{title}\' should be one of: {types}'.format(type=elem.get("type"), title=elem.get("title"), types=valid_types)
```